### PR TITLE
add pdb to certmanager, galley, gateways, mixer and pilot

### LINKED ISSUE #9951: Pod Disruption Budget for helm install
**Describe the feature request**
As there's no pdb in helm chart, when you upgrade istio chart, existing pods turns into `Terminating` status before new pods become healthy which causes downtime! I temporally set `replicaCount: 2` to contain healthy pod, but I don't think this is right solution. Also, istio system downtime may cause without pdb in various situations like node draining.
I suggest to add PodDisruptionBudget templates for istio components.

**Describe alternatives you've considered**
Setting `replicaCount: 2`, but this doesn't seem to be right solution.

**Additional context**
Eviction like node draining(`kubectl drain`) can also cause istio downtime. As istio manages external endpoints, those downtime situations are critical.

### PR DESCRIPTION
**Pull request by kimxogus**
closes #9951

### CODE REVIEW

**Comment by sdake** [comment]
@kimikowang looks like this bug has been prioritized for the istio 1.1 release.  Could you please rebase against the release-1.1 branch?

thanks!
-steve

File: install/kubernetes/helm/subcharts/certmanager/values.yaml [hunk 1]
```diff
@@ -7,3 +7,8 @@ enabled: false
 hub: quay.io/jetstack
 tag: v0.5.0
 resources: {}
+
+podDisruptionBudget:
+  enabled: true
+  minAvailable: 1
+  # maxUnavailable: 0
```
Comment (linsun) [inline install/kubernetes/helm/subcharts/certmanager/values.yaml]: Do you see any reason why an istio component would want to expose this maxUnavailable attribute?

**Comment by linsun** [comment]
@kimxogus thanks for the PR and introduce pdb to us.  Please rebase the PR to release-1.1 as @sdake has suggested so we can get it merged.

LGTM in general.  I have some questions:
- maxUnavailable, i am having a hard time to see how it can apply to istio control plane components like pilot, citadel, etc.  Seems mixAvailable is sufficient?
- What do you think of exposing this as a global config?  Is there any reason why user would want to control this different for each control plane component?
- What do you think of enabling this thing by default for istio control plane components?  Seems a useful thing to have it on.

File: install/kubernetes/helm/subcharts/certmanager/values.yaml (hunk 1 above)
Comment (kimxogus) [reply install/kubernetes/helm/subcharts/certmanager/values.yaml]: I just followed general convention in https://github.com/helm/charts 
This pattern is generally used in many charts. no specific reason for this.
I changed them with `global.defaultPodDisruptionBudget` like `global.defaultResources`

**Comment by kimxogus** [comment]
@linsun Rebased to release-1.1

- maxUnavailable, i am having a hard time to see how it can apply to istio control plane components like pilot, citadel, etc. Seems mixAvailable is sufficient?
    - Generally `minAvailable` would be sufficient, but with default `values.yaml`, replica counts of istio components will be 1. This might cause some problem in some cases like node draining. If user can tolerate short downtime, maybe `maxUnavailable: 1` and unset `minAvailable` can be accepted.
- What do you think of exposing this as a global config? Is there any reason why user would want to control this different for each control plane component?
    - Good suggestion! I added them as `.Values.global.defaultPodDisruptionBudget` just like `.Values.global.defaultResources`.
- What do you think of enabling this thing by default for istio control plane components? Seems a useful thing to have it on.
    - Enabled them by default.

**Comment by kimxogus** [comment]
Are those test failures related to helm chart?

**Comment by linsun** [comment]
fixes #9951

testing auto close :)

**Pull request merged**